extensions/everhour/src/views/TimeSubmitForm.tsx

```diff
@@ -0,0 +1,37 @@
+import { Form, ActionPanel, SubmitFormAction, showToast, ToastStyle, useNavigation } from "@raycast/api";
+import { submitTaskHours } from "../api";
+
+export function TimeSubmitForm({ taskId, fetchTasks }: { taskId: string; fetchTasks: () => Promise<void> }) {
+  const { pop } = useNavigation();
+
+  const handleSubmit = async ({ hours }: { hours: string }) => {
+    try {
+      const { taskName } = await submitTaskHours(taskId, hours);
+
+      if (taskName) {
+        await fetchTasks();
+        pop();
+        await showToast(
+          ToastStyle.Success,
+          `Added ${hours} ${parseInt(hours) === 1 ? "hour" : "hours"} to ${taskName}`
+        );
+      } else {
+        await showToast(ToastStyle.Failure, "Failed to add time");
+      }
+    } catch (error) {
+      await showToast(ToastStyle.Failure, "Failed to add time");
+    }
+  };
```

Comment:
We show an animated toast for long running operations. This way the user knows that something is happening. You could do something like this:

```suggestion
  const handleSubmit = async ({ hours }: { hours: string }) => {
    const toast = await showToast(ToastStyle.Animated, "Adding time")
    
    try {
      const { taskName } = await submitTaskHours(taskId, hours);

      if (taskName) {
        await fetchTasks();
        pop();
        
        toast.style = ToastStyle.Success;
        toast.title = "Added time"
        toast.message = `Added ${hours} ${parseInt(hours) === 1 ? "hour" : "hours"} to ${taskName}`;
      } else {
        toast.style = ToastStyleFailure;
        toast.title = "Failed adding time";
      }
    } catch (error) {
      toast.style = ToastStyle.Failure;
      toast.title = "Failed adding time";
      toast.message = error instanceof Error ? error.message : undefined;
    }
  };
```

You can apply this pattern also to your other async updates.